base/intfuncs.jl

```diff
@@ -234,6 +234,15 @@ const HWNumber = Union{HWReal, Complex{<:HWReal}, Rational{<:HWReal}}
 @inline literal_pow(::typeof(^), x::HWNumber, ::Val{2}) = x*x
 @inline literal_pow(::typeof(^), x::HWNumber, ::Val{3}) = x*x*x
 
+# don't use the inv(x) transformation here since x^p is slightly more accurate
```

Comment:
It seems like you should have to opt-in to getting a less accurate answer that violates referential transparency, rather than opt-out of it.

Reply:
For most types, it's not less accurate. For most floating-point-based types (complex numbers, matrices, etc.), negative integer powers are already computed by `inv(x)^-n`.    And for many other types (e.g. `Int`), negative literal powers wouldn't be defined at all without this transformation.

Can we not re-litigate the whole `literal_pow` debate in this PR?   The question is whether this PR is an improvement on the current `literal_pow` rules.